eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/eventdetails/BasicEventDetails.java

```diff
@@ -55,54 +63,48 @@ public BasicEventDetails(@NonNull Event event)
     @Override
     public void show(@NonNull final Activity activity)
     {
-        new Thread(new StartEventDetailsRunnable(activity, mEvent)).start();
-    }
-
-
-    /**
-     * A {@link Runnable} that starts the details activity in the background after trying to load any actions from the cache.
-     */
-    private final static class StartEventDetailsRunnable implements Runnable
-    {
-        private final Activity mActivity;
-        private final Event mEvent;
-
+        final OptionalArgument<MicroFragmentHost> host = new OptionalArgument<>(Keys.MICRO_FRAGMENT_HOST, activity);
 
-        private StartEventDetailsRunnable(@NonNull Activity activity, @NonNull Event event)
+        if (!host.isPresent())
         {
-            mActivity = activity;
-            mEvent = event;
+            throw new UnsupportedOperationException("Opening EventDetails with already loaded Event, from non-host Activity is not implemented.");
         }
 
-
-        @Override
-        public void run()
+        final Timestamp timestamp = new UiTimestamp();
+        new Thread(new Runnable()
         {
-            Intent intent = new Intent(mActivity, MicroFragmentHostActivity.class);
-            Bundle nestedBundle = new Bundle();
-
-            FutureServiceConnection<ActionService> actionService = new ActionService.FutureConnection(mActivity);
-            try
+            @Override
+            public void run()
             {
-                List<Link> actions = actionService.service(40).cachedActions(mEvent.uid());
-                // Start the details with the actions from the cache.
-                nestedBundle.putParcelable("MicroFragment", new ShowEventMicroFragment(mEvent, actions));
-            }
-            catch (InterruptedException | TimeoutException | NoSuchElementException e)
-            {
-                // An error occurred or the actions are not in the cache yet, let the details activity load the actions.
-                nestedBundle.putParcelable("MicroFragment", new ActionLoaderMicroFragment(mEvent));
+                Optional<List<Link>> actions = loadActionsFromCache(new ActionService.FutureConnection(activity));
+
+                MicroFragment microFragment = actions.isPresent() ?
+                        new ShowEventMicroFragment(mEvent, actions.value()) : new ActionLoaderMicroFragment(mEvent);
+
+                host.value().execute(activity, new Swiped(new ForwardTransition<>(microFragment, timestamp)));
             }
-            finally
+
+
+            private Optional<List<Link>> loadActionsFromCache(FutureServiceConnection<ActionService> actionService)
             {
-                if (actionService.isConnected())
+                try
+                {
+                    List<Link> actions = actionService.service(40).cachedActions(mEvent.uid());
```

Comment:
please get rid of the temp variable. You can easily put this into a one-liner without reducing readability.